components/OpenaiFeedback.js

```diff
@@ -0,0 +1,102 @@
+import useSWR from "swr";
+import { useState } from "react";
+import styled from "styled-components";
+import InfoIcon from "../assets/buttons/info-circle.svg";
+
+export default function OpenaiFeedback({ emotion }) {
+  const { mutate, isLoading } = useSWR(`/api/emotionEntries/${emotion._id}`);
+  const [showFeedback, setShowFeedback] = useState(false);
+  const [apiOutput, setApiOutput] = useState(emotion.openaiFeedback || "");
+
+  async function handleGenerateFeedback() {
+    try {
+      const response = await fetch(
+        `/api/emotionEntries/${emotion._id}/feedback-generation`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(emotion),
+        }
+      );
+
+      if (response.ok) {
+        const data = await response.json();
+        setApiOutput(data.output);
+        mutate();
+        setShowFeedback(true);
+      } else {
+        console.error("Feedback generation failed:", await response.text());
+      }
+    } catch (error) {
+      console.error("Error during feedback generation:", error.message);
+    }
+  }
+
+  function toggleFeedback() {
+    setShowFeedback(!showFeedback);
+  }
+
+  return (
+    <>
+      <StyledContainer>
+        <StyledButton
+          isVisible={showFeedback} // Prop übergeben
```

Comment:
You should prefix props you pass to styled component with "$", otherwise you will get an error message in the terminal:

```suggestion
          $isVisible={showFeedback} // Prop übergeben
```